packages/alfa-style/src/element/predicate/has-transparent-background.ts

```diff
@@ -0,0 +1,64 @@
+import { Cache } from "@siteimprove/alfa-cache";
+import { Color, Keyword } from "@siteimprove/alfa-css";
+import { Device } from "@siteimprove/alfa-device";
+import { Element } from "@siteimprove/alfa-dom";
+import { Context } from "@siteimprove/alfa-selector";
+import { Style } from "../../style";
+import { Predicate } from "@siteimprove/alfa-predicate";
+
+const { isReplaced, isElement } = Element;
+
+
+const { or, test } = Predicate;
+
+const cache = Cache.empty<Device, Cache<Context, Cache<Element, boolean>>>();
+
+/**
+ * @public
+ */
+
+export function hasTransparentBackground(
+  device: Device,
+  context: Context = Context.empty()
+): Predicate<Element> {
+  return (element) =>
+    cache
+      .get(device, Cache.empty)
+      .get(context, Cache.empty)
+      .get(element, () => {
+        if (
+          test(
+            or(
+              isReplaced,
```

Comment:
Let's add some comments.
```suggestion
              // Replaced elements are assumed to be replaced by non-transparent content.
              isReplaced,
```